genomesubsampler/prototypeSelection.py

```diff
@@ -62,11 +109,10 @@ def distance_sum(elements, dm):
     function signature with type annotation for future use with python >= 3.5
     def distance_sum(elements: Sequence[str], dm: DistanceMatrix) -> float:
     '''
-
     return np.tril(dm.filter(elements).data).sum()
 
 
-def prototype_selection_exhaustive(dm, num_prototypes,
+def prototype_selection_exhaustive(dm, num_prototypes, seedset=[],
```

Comment:
Please use `seedset=None` and test whether it is `None` in the body of the function. It is a dangerous pattern to set a default object like this, and can lead to unexpected behavior:

```python
>>> def foo(bar=[]):
...     bar.append('a')
...     print(bar)
...
>>> foo()
['a']
>>> foo()
['a', 'a']
>>> foo()
['a', 'a', 'a']
>>> foo()
['a', 'a', 'a', 'a']
```